uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/CRM/HRLeaveAndAbsences/Service/WorkPatternCalendarTest.php

```diff
@@ -67,24 +68,18 @@ public function testGetShouldUseTheDefaultWorkPatternIfTheContactHasNoActiveWork
     // middle, assuming all the other will be correct.
     $this->assertCount(15, $calendarDates);
     $this->assertEquals('2015-01-05', $calendarDates[0]['date']);
-    $this->assertEquals($workDayTypes['working_day'], $calendarDates[0]['type']);
+    $this->assertEquals($workDayTypes['working_day']['value'], $calendarDates[0]['type']);
```

Comment:
The whole point of having this `getWorkDayTypeOptionsArray()` helper was to get a list of option values in the exact same form as the one returned by the calendar, which had a lot of more data than a simple `buildOptions()` would be able to return. With the change in the response format, the `type` property only contains the value, and now, I believe, `buildOptions()` can be used again. So here is what I think we should do:

- Replace all the calls to `getWorkDayTypeOptionsArray()` with `WorkDay::buildOptions('type_id', 'validate')`
- Remove all these `['value']` keys you've added
- Remove the `CRM_HRLeaveAndAbsences_WorkPatternHelpersTrait` completely, as it seems it's only used by the Work Pattern tests